src/GoodList.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class GoodsList extends React.Component {
+  state = {
+    goodsCopy: [...this.props.goods],
+    select: 1,
+  }
+
+  reverseGoods = () => {
+    this.setState(({ goodsCopy }) => ({
+      goodsCopy: [...goodsCopy].reverse(),
+    }));
+  }
+
+  sortGoods = () => {
+    this.setState(({ goodsCopy }) => ({
+      goodsCopy: [...goodsCopy].sort(),
+    }));
+  }
+
+  resetGoods = () => {
+    this.setState(() => ({
+      goodsCopy: [...this.props.goods],
+      select: 1,
+    }));
+  }
+
+  sortByLengthGoods = () => {
+    this.setState(({ goodsCopy }) => ({
+      goodsCopy: [...goodsCopy].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  sortBySelectedLength = (event) => {
+    const { value } = event.target;
+
+    this.setState({
+      select: value,
+      goodsCopy: [...this.props.goods].filter(el => el.length >= value),
```

Comment:
```suggestion
      goodsCopy: this.props.goods.filter(el => el.length >= value),
```
filter и так возвращает копию массива